### PR TITLE
Edit/Delete Reviews

### PR DESCRIPTION
**Pull request by jeremynguyen593**
- Users now able to edit and delete reviews
- Editing/deleting updates the values of the averages
- Icons are bottom left
- User's review is on top of the list
- Editing restores state of the old review
- Created a loading screen to rerender after deleting and update (since you can't navigate to the same route and rerender)

Testing:
- Editing and deleting works/updates the values
- Loading screen in between operations
- Editing restores old values
- User's review is on top
- Signed out users cannot see icons and see them in the correct order

### CODE REVIEW

File: app/src/Rating/Rating.tsx [hunk 1]
```diff
@@ -111,15 +115,15 @@ export const ReviewHolder: React.FC<{classNum : string}> = ({classNum}) => {
...
-                <RatingScale category={2} setReview={setRatingContents}/>
-                <RatingScale category={3} setReview={setRatingContents}/>
+                <RatingScale category={1} setReview={setRatingContents} initialValue={initialState.rating_one}/>
+                <RatingScale category={2} setReview={setRatingContents} initialValue={initialState.rating_two}/>
+                <RatingScale category={3} setReview={setRatingContents} initialValue={initialState.rating_three}/>
             </div>
             <div className="other-inputs">
-                <InputField setReview={setRatingContents} field="Quarter Taken"/>
-                <InputField setReview={setRatingContents} field="Professor"/>
+                <InputField setReview={setRatingContents} initialValue={reviewState !== undefined ? initialState.quarter : ""} field="Quarter Taken"/>
```
Comment (thaowii) [inline app/src/Rating/Rating.tsx:123]: I would put these parameters on separate lines to improve readability

File: app/src/utils/LoadingPage.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+        return () => clearTimeout(timeout);
+    }, [navigate]);
+
+    return (
+        <div className="innerpage">
+        <div className="loading-spinner center" />
+        <div className="loading-text">Loading...</div>
+        </div>
+  ) ;
```
Comment (thaowii) [inline app/src/utils/LoadingPage.tsx:22]: Indentation off

File: app/src/utils/LoadingPage.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React, {useEffect} from 'react';
+import {useParams, useNavigate} from 'react-router-dom';
+
+const LoadingPage: React.FC = () => {
```
Comment (thaowii) [inline app/src/utils/LoadingPage.tsx:4]: Could you do export const instead of having the export default at the bottom?

**Review by thaowii — CHANGES_REQUESTED** [review]

**Review by celestinebuendia — DISMISSED** [review]
Lot of work! Looks good.

File: app/src/utils/LoadingPage.tsx (hunk 3 above)
Comment (jeremynguyen593) [reply app/src/utils/LoadingPage.tsx:4]: Fixed

File: app/src/utils/LoadingPage.tsx (hunk 2 above)
Comment (jeremynguyen593) [reply app/src/utils/LoadingPage.tsx:22]: Fixed

File: app/src/Rating/Rating.tsx [hunk 4]
```diff
@@ -37,8 +40,8 @@ export const ReviewPage: React.FC = () => {
                 </div>
             )}
             <div className="page-contents">
-                <ReviewHolder classNum={classNum!}/>
-                <div className="rating-instr">
+            <ReviewHolder classNum={classNum!} reviewState={newReviewState} />                
+            <div className="rating-instr">
```
[the remark this answers is not in the document]
Comment (jeremynguyen593) [reply app/src/Rating/Rating.tsx:44]: Fixed

File: app/src/Rating/Rating.tsx (hunk 1 above)
Comment (jeremynguyen593) [reply app/src/Rating/Rating.tsx:123]: Fixed

**Review by thaowii — APPROVED** [review]

**Pull request merged**